source/ways-of-working/guides/flyway-database-migrations.html.md.erb

```diff
@@ -0,0 +1,137 @@
+---
+title: Flyway database migration
+last_reviewed_on: 2022-03-17
+review_in: 6 months
+weight: 10
+---
+
+# <%= current_page.data.title %>
+
+To prevent application startup failure for larger migrations due to Kubernetes probes it's essential that databases are migrated outside of Kubernetes via the Continuous Delivery pipeline.
+This page attempts to show the steps required to perform a safe database migration.
+
+For a more detailed explanation on why this is necessary see: [Making a database schema change](../standards/db-schema-change.html)
+
+## How to apply Flyway migration as part of pipeline
+
+Example [pull request](https://github.com/hmcts/am-role-assignment-service/pull/1149)
+
+Alternatively see below for a step-by-step guide.
+
+### Config
+
+Enable database migration for Jenkins:
+
+- Add `enableDbMigration('<PROJECT_NAME>')` to `Jenkinsfile_CNP` file
+
+Add correct config to `build.gradle`:
+
+- Ensure the [latest flyway version](https://plugins.gradle.org/plugin/org.flywaydb.flyway) is set e.g. `id 'org.flywaydb.flyway' version '<VERSION_NUMBER>'`
+- Add this block:
+
+```groovy
+flyway {
+  url = System.getenv('FLYWAY_URL')
+  user = System.getenv('FLYWAY_USER')
+  password = System.getenv('FLYWAY_PASSWORD')
+  baselineOnMigrate = true
+  baselineVersion = '000'
+}
+
+task migratePostgresDatabase(type: org.flywaydb.gradle.task.FlywayMigrateTask) {
+  baselineOnMigrate = true
+  if (project.hasProperty("dburl")) {
+    url = "jdbc:postgresql://${dburl}"
+  }
+}
+```
+
+Ensure the `RUN_DB_MIGRATION_ON_STARTUP` is configured appropriately:
+
+- Add `RUN_DB_MIGRATION_ON_STARTUP: true` to the `values.preview.template.yaml` file
+- Add `RUN_DB_MIGRATION_ON_STARTUP: false` to the `values.yaml` file - This is so that preview will migrate on startup
```

Comment:
wrong way around
```suggestion
- Add `RUN_DB_MIGRATION_ON_STARTUP: true` to the `values.preview.template.yaml` file - This is so that preview will migrate on startup
- Add `RUN_DB_MIGRATION_ON_STARTUP: false` to the `values.yaml` file
```